### ISSUE TITLE
How can I add layer normalization to the keras features?

### ISSUE THREAD

**Issue by rylanchiu** [fchollet/keras#3519]
As far as I know, keras has batch normalization feature. But I want the feature of layer normalization, which is suggested by this [paper](https://arxiv.org/pdf/1607.06450v1.pdf). In general, it is to normalize the inputs to neurons in every layers. As a result, the neurons in the same layer share equal mean and variance, which is helpful to the covariate shift problem. Now I am going to implement this feature myself. If possible, I can also PR to do some contribution for keras. I wonder how and where should I add this feature. Should I write one more layer class like `Batch Normalization` ? Or I can just normalize the input during feedforward?  Thank you in advance!

**Comment by abishekk92** [comment]
I came here looking to open a similar issue. Nice, to see someone already thinking about it for Keras. Here is a reference [implementation](https://github.com/ryankiros/layer-norm/blob/master/torch_modules/LSTM_LN.lua) in torch from one of the authors of the Layer Normalization paper. I think, it would be useful to have `LayerNormalization` layer available individually as a layer as it fits nicely with treating each layer like a lego block that people can compose.

**Comment by rylanchiu** [comment]
@abishekk92  It's great that someone has the same idea! I have implemented it by customize the `Dense` layer (as I wanted to use it on the fully connected layer). I think the implementation of layer normalization itself could be quite easy. The most tricky thing should be the initialization of gamma. According to others and my experiments, the initial value of gamma matters a lot.

**Comment by Kyubyong** [comment]
I agree. By the way, what's the difference between layer normalization and the mode=1 in BatchNormalization?

**Comment by happygds** [comment]
Hi, I also think that the layer normalization equals the mode=1 in BatchNormalization. 
Then could we use TimeDistributed(BatchNormalization(mode=1) for recurrent neural network? This would be quite convenient !!!
